packages/core/lib/modules/data-migration.js

```diff
@@ -27,6 +28,11 @@ export async function migrate({wallet}: MigrateParams) {
 
   Logger.debug(`Wallet version ${version}`);
 
+  if (version !== currentWalletVersion) {
+    const snapshot = await getStorage().getItem(wallet.walletId);
+    await getStorage().setItem(`${wallet.walletId}-snapshot`, snapshot);
```

Comment:
how do we want to handle this blowing up? Do we catch it and continue on? Let the app crash? But if we let it crash do we get into a state where it will crash forever?

Reply:
If it crashes, the `migrated` event will not be fired and the app will stay in the splash screen forever

Reply:
we will have sentry logs for that, but not sure if that would make sense to display the wallet UI using data in a wrong format

Reply:
would create a lot of other issues probably